### PR TITLE
require-dev support for ORM 3

### PR DESCRIPTION
**Pull request by dmaicher**
## Subject

I am targeting this branch, because its BC.

Part of https://github.com/sonata-project/dev-kit/issues/2674

## Changelog

```markdown
### Added
- compatibility with ORM 3
```

### CODE REVIEW

File: tests/App/Entity/Media.php [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 #[ORM\Entity]
 #[ORM\Table(name: 'media__media')]
+#[ORM\AttributeOverrides([
+    new ORM\AttributeOverride(name: 'length', column: new ORM\Column(type: Types::DECIMAL, precision: 2, nullable: true)),
```
Comment (dmaicher) [inline tests/App/Entity/Media.php]: fails with PHP 8.0 :vomiting_face: 

Can we just update it here?

https://github.com/sonata-project/SonataMediaBundle/blob/e0436ccf5153e260a70d224930dd18da42682478/src/Resources/config/doctrine/BaseMedia.orm.xml#L17

Not sure how this mapping is used :man_shrugging:

Comment (dmaicher) [reply tests/App/Entity/Media.php]: Adding `precision` is required with ORM 3 for decimal columns

Comment (dmaicher) [reply tests/App/Entity/Media.php]: Will be fixed by https://github.com/sonata-project/SonataMediaBundle/pull/2448

Comment (VincentLanglet) [reply tests/App/Entity/Media.php]: done https://github.com/sonata-project/SonataMediaBundle/releases/tag/4.13.0

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**